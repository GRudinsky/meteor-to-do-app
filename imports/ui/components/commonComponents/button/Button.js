import React from 'react';
import './Button.scss';

export const Button = ({ ...props }) => {
  const { clickHandler, disabled, id, secondary, text } = props;
  const modifierClass = secondary ? 'secondary' : 'primary';

  return (
    <>
      <button
        id={ id }
        className={ `button button--${ modifierClass }` }
        onClick={ clickHandler }
        disabled={ disabled }
      >
        { text }
      </button>
    </>
  );
};

export default Button;
