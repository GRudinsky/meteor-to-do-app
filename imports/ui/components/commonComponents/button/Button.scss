@import '../../../../styles/colours.scss';

.button {
  font-size: 1em;
  font-family: 'Montserrat', sans-serif;
  font-weight: bold;
  min-width: 100px;
  height: 30px;
  border-radius: 3px;
  color: $c-white;
  background-color: $c-black;
  border: 1px solid $c-black;
  box-shadow: 0 3px 3px $c-black;
  cursor: pointer;

  &--secondary {
    color: $c-black;
    background-color: $c-white;
  }

   &:focus {
    outline: 0;
  }
}