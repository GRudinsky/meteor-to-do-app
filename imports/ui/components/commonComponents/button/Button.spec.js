import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';

describe('<Button />',() => {
  let wrapper,
    props = {
      clickHandler: jest.fn(),
      text: 'SomeText',
      id: 'someId',
    };

  beforeEach(() => {
    wrapper = mount(<Button { ...props }/>);
  });

  afterEach(() => {
    wrapper = null;
  });

  it('should render correctly',() => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call clickHandler on click',() => {
    
    wrapper.find('button').simulate('click');

    expect(props.clickHandler).toHaveBeenCalledTimes(1);
  });

  [{ secondary: true }, { primary: true }].forEach(item => {
    const className = Object.keys(item)[0];
    
    it(`should have button--${ className } class when passed ${ className } prop`,() => {
      
      const newProps = {
        ...props,
        ...item
      };

      wrapper = mount(<Button { ...newProps } />);

      expect(wrapper.find(`button.button--${ className }`)).toHaveLength(1);
    });
  });
});