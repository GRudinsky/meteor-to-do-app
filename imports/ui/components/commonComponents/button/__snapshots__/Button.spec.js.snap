// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<Button
  clickHandler={[MockFunction]}
  id="someId"
  text="SomeText"
>
  <button
    className="button button--primary"
    id="someId"
    onClick={[MockFunction]}
  >
    SomeText
  </button>
</Button>
`;
